### DISCUSSION TITLE
Optional arguments with default values

### DISCUSSION THREAD

**Discussion by patelnets** [denoland/deno#10195] [General]
In the style guide docs an options object to place all optional parameters is regarded as good practice https://deno.land/manual@v1.5.4/contributing/style_guide#exported-functions-max-2-args-put-the-rest-into-an-options-object 

 The docs don't expand how one may go about using default values to these optional parameters. 

Take the use case here 

```
const foo = (
  requiredArg1: string,
  options: {
    optionalArgument1?: string;
    optionalArgument2?: string;
  } = {
    optionalArgument1: 'defaultValueForOptionalArgument1',
    optionalArgument2: 'defaultValueForOptionalArgument2',
  },
): void => {
  console.log('requiredArg1', requiredArg1);
  console.log('options.optionalArgument1', options.optionalArgument1);
  console.log('options.optionalArgument2', options.optionalArgument2);
};
```

Calling this like    
```
foo('passedInRequiredValue', {
      optionalArgument1: 'passedInOptionalArgument1',
    });
```

Gives this output
```
requiredArg1 passedInRequiredValue
options.optionalArgument1 passedInOptionalArgument1
options.optionalArgument2 undefined
```

Even though I understand why the output is such it seems a very easy thing to miss and if I were to look over the code quickly I would expect the below
 
```
requiredArg1 passedInRequiredValue
options.optionalArgument1 passedInOptionalArgument1
options.optionalArgument2 defaultValueForOptionalArgument2
```

Taking this into account I was wondering if Deno developers still think the options pattern when used with default values is a good style. I know there are arguments for both for and against but I would like to hear your thoughts 😃 

Thanks!

**Comment by nayeemrmn** [comment]
The correct way is:
```ts
function foo(
  requiredArg1: string,
  {
    optionalArgument1 = "defaultValueForOptionalArgument1";
    optionalArgument2 = "defaultValueForOptionalArgument2";
  } = {},
): void {
  console.log("requiredArg1", requiredArg1);
  console.log("options.optionalArgument1", options.optionalArgument1);
  console.log("options.optionalArgument2", options.optionalArgument2);
};
```

We should show an example in the style guide.
